[llvm-opt-report] Show scalable vectorization factors

iamlouk opened the pull request:
Scalable vectorization factors are printed as "vscale x VF" where VF is the known minimum number of elements, a integer. Currently, llvm-opt-report always expects a integer (like for vectorization with fixed-sized vectors), and does not display any vectorization factor in the output (just 'V', but without a number).

This patch adds support for scalable vectorization factors and prints them as "VNx<VF>", so for example "VNx4". The "Nx" is used to differentiate between fixed-sized and scalable factors, and is consistent with the way LLVM mangles scalable vectors in other places.

iamlouk commented:
Hello, there seams to be no fixed maintainer assigned to llvm-opt-report, @francisvm sorry to ping you, but could you have a look at this? Thank you very much in advance.

francisvm reviewed (approved):
No bother at all, LGTM!

iamlouk committed:
[llvm-opt-report] Show scalable vectorization factors

Scalable vectorization factors are printed as "vscale x VF" where VF is
the known minimum number of elements, a integer. Currently,
llvm-opt-report always expects a integer (like for vectorization with
fixed-sized vectors), and does not display any vectorization factor
in the output (just 'V', but without a number).

This patch adds support for scalable vectorization factors and prints
them as "VNx<VF>", so for example "VNx4". The "Nx" is used to
differentiate between fixed-sized and scalable factors, and is
consistent with the way LLVM mangles scalable vectors in other places.

iamlouk commented:
@francisvm  I don't have write access, would you mind suggesting other reviewers or merging this PR yourself? Thanks!

iamlouk commented:
@nikic , as I can't find another still-active contributor and there is no maintainer assigned to this, would you mind taking a look at the PR (and in case you approve, click the merge button for me?). Thank you very much in advance, sorry for the disturbance.
